packages/tf-next/src/index.ts

```diff
@@ -1,26 +1,111 @@
+import cuid from 'cuid';
 import yargs from 'yargs';
 
-yargs.command(
-  'build',
-  'Build the next.js project',
-  (yargs_) => {
-    return yargs_
-      .option('skipDownload', {
-        type: 'boolean',
-        description: 'Runs the build in the current working directory',
-      })
-      .option('verbose', {
-        type: 'boolean',
-        description: 'Run with verbose logging',
+// TODO: Have a central configuration for AWS API versions
+// TODO: Handle trying to delete a deployment that doesn't exist
+// TODO: Handle trying to create a deployment with an id that already exists
+
+yargs
+  .scriptName('tf-next')
+  .usage('$0 <cmd> [args]')
+  .command(
+    'build',
+    'Build the next.js project',
+    (yargs_) => {
+      return yargs_
+        .option('skipDownload', {
+          type: 'boolean',
+          description: 'Runs the build in the current working directory',
+        })
+        .option('verbose', {
+          type: 'boolean',
+          description: 'Run with verbose logging',
+        });
+    },
+    async ({ skipDownload, verbose }) => {
+      const cwd = process.cwd();
+
+      (await import('./commands/build')).default({
+        skipDownload,
+        logLevel: verbose ? 'verbose' : 'none',
+        cwd,
+      });
+    }
+  )
+  .command(
+    'list-deployments',
+    'List existing deployments',
+    async () => {
+      (await import('./commands/list-deployments')).default({});
+    }
+  )
+  .command(
+    'create-deployment',
+    'Create a new deployment that runs in parallel to the existing deployments',
```

Comment:
Runs when? Does creating it include triggering a first deployment?

Reply:
This is supposed to be run, after TF was run to create the platform infra, to dynamically create deployments. Running this will not create the platform infra, so there's a dependency there that is codified through some error messages in the command. Basically this would be used for example in a GH action to deploy a preview environment for a branch. This would likely mean a new deployment per update of the branch, because we (currently) have no way to track what the deployment for a branch is. But I think that's how Vercel does it, too? They then just helpfully create a subdomain alias for the branch and point it to the new deployment. ping @ofhouse

Reply:
Exactly. Vercel does per commit deployments and assigns the alias domain (e.g. branchname.example.com) for the branch to the latest commit.

Reply:
So i think the description of the command in the CLI for users can be improved.